speechbrain/lobes/models/huggingface.py

```diff
@@ -0,0 +1,361 @@
+        and the HuggingFace Pretrain Object. It is necessary to perform a custom
+        loading because HuggingFace adds a level to the checkpoint when storing
+        the model breaking the compatibility Pretrain and model de/serialization.
+
+        For example, a typical HuggingFaceWav2Vec2 checkpoint for a given parameter
+        would be: model.conv.weight.data while for HuggingFaceWav2Vec2Pretrain it
+        is: model.wav2vec2.weight.data (wav2vec2 must be removed before loading).
+        """
+        if modify_state_dict_partial_fn is not None:
+            modified_state_dict = modify_state_dict_partial_fn(path)
+        else:
+            modified_state_dict = torch.load(path, map_location="cpu")
+
+        incompatible_keys = self.model.load_state_dict(
+            modified_state_dict, strict=False
+        )
+        for missing_key in incompatible_keys.missing_keys:
+            logger.warning(
+                f"During parameter transfer to {self.model} loading from "
+                + f"{path}, the transferred parameters did not have "
+                + f"parameters for the key: {missing_key}"
+            )
+        for unexpected_key in incompatible_keys.unexpected_keys:
+            logger.warning(
+                f"The param with the key: {unexpected_key} is discarded as it "
+                + "is useless for finetuning this HuggingFaceTransformer."
+            )
+
+    def forward(self, **kwargs):
+        """Users should modify this function according to their own tasks."""
+        raise NotImplementedError
+
+    def forward_encoder(self, **kwargs):
+        """Users should modify this function according to their own tasks."""
+        raise NotImplementedError
+
+    def forward_decoder(self, **kwargs):
+        """Users should modify this function according to their own tasks."""
+        raise NotImplementedError
+
+    def decode(self, **kwargs):
+        """Might be useful for models like mbart, which can exploit SB's beamsearch for inference
+        Users should modify this function according to their own tasks."""
+        raise NotImplementedError
+
+    def encode(self, **kwargs):
+        """Customed encoding for inference
+        Users should modify this function according to their own tasks."""
+        raise NotImplementedError
+
+    def freeze_model(self, model):
+        """
+        Freezes parameters of a model.
+        This should be overrided too, depending on users' needs, for example, adapters use.
+
+        Arguments
+        ---------
+        model : from AutoModel.from_config
+            Valid HuggingFace transformers model object.
+        """
+        model.eval()
+        for param in model.parameters():
+            param.requires_grad = False
+
+    def override_config(self, config):
+        """Users should modify this function according to their own tasks."""
+        return config
+
+
+def make_padding_masks(src, wav_len=None, pad_idx=0):
```

Comment:
Don't we have a function in SpeechBrain to do this already? I believe we do, maybe in util, or attention or somewhere.